revision_utils.py

```diff
@@ -1,21 +1,53 @@
 # -*- coding: utf-8 -*-
 """Utility functions for revision management."""
 
-__copyright__ = 'Copyright (c) 2019-2023, Utrecht University'
+__copyright__ = 'Copyright (c) 2019-2024, Utrecht University'
 __license__   = 'GPLv3, see LICENSE'
 
 
 import datetime
+import hashlib
 import os
 
 from revision_strategies import get_revision_strategy
 from util import constants, log
 
 
-def calculate_end_of_calendar_day(ctx):
-    """Calculate the unix timestamp for the end of the current day (Same as start of next day).
+def revision_eligible(max_size, data_obj_exists, data_size, path, groups, revision_store_exists):
```

Comment:
Can we also incorporate the collection name and data object name checks in this function? They are currently in https://github.com/UtrechtUniversity/yoda-ruleset/blob/1886415e0b17c6e96cbf4d279a2364544f873d7c/revisions.py#L261-L266

The advantages of moving these checks here would be:
1. We have all the checks for seeing if a data object is eligible for revisions in one place.
2. We can add a unit tests or two for these checks
3. We can perform these checks both at AVU creation time and at revision creation time, so that we also cover situations where a data object has been moved / renamed after AVU creation, for example if a data object has been moved to trash shortly after being created.